lib/lbt/bundle/Builder.js

```diff
@@ -357,6 +361,29 @@ class BundleBuilder {
 		sequence.sort();
 	}
 
+	addSourceMap(contentMap, resource) {
+		if (contentMap != null) {
+			const map = new sourceMap.SourceMapConsumer(contentMap);
+			map.eachMapping((mapping) => {
+				if (mapping.source) {
+					this._sourceMap.addMapping({
+						generated: {
+							line: this.outW.lineOffset + mapping.generatedLine,
+							column: (mapping.generatedLine === 1 ? this.outW.columnOffset : 0) + mapping.generatedColumn
+						},
+						original: mapping.originalLine == null ? null : {
+							line: mapping.originalLine,
+							column: mapping.originalColumn
+						},
+						source: mapping.originalLine != null ?
+							(path.join(path.relative(path.dirname(this._sourceMap._file), path.dirname(resource.name)), this.optimize ? ModuleName.getDebugName(path.basename(resource.name)) : path.basename(resource.name))).replace(/\\/g, "/") : null,
```

Comment:
the special handling for \\ looks very suspicious to me. AFAIK, we do this nowhere.

And line length won't make @RandomByte happy.

Reply:
Well i'm not sure how you deal with the path operation so far, but I couldn't find anything specific.
In my case I need to get relative file path for the sources, usually this would be done with `path` directly but since you have a vfs behind the scene all file paths are unix like (which in turn break the usage of path on windows).
I would love for a better alternative (which doesn't involve rewriting basic functionalities of node)

Reply:
> since you have a vfs behind the scene all file paths are unix like (which in turn break the usage of path on windows)

When working with paths coming from or passed to our vfs, use [`path.posix`](https://nodejs.org/api/path.html#path_path_posix). E.g. `path.posix.relative`.

In cases were you have a native path and want to use it with the vfs, we use the [slash](https://github.com/sindresorhus/slash) module: https://github.com/SAP/ui5-builder/blob/9d213ced942ed7832fbb7b50f9d444f441941f35/lib/types/application/ApplicationFormatter.js#L41-L46